Fix: Windows terminal cell size detection fallback

Vedant-Asati03 opened the pull request:
Problem:
- Terminal cell size detection fails on Windows when using ANSI escape sequences(Some older terminals like like CMD do not understand this ANSI escape sequence `\x1b[16t`)
- Application crashes with TerminalError when detection fails

Solution:
- Skip ANSI escape sequence method on Windows platforms
- Use default VT340 cell sizes (10x20) as fallback
- Improve error handling to prevent application crashes
- Add better debug logging

This fixes the TerminalError: "Failed to get cell size" on Windows systems.

lnqs commented:
Thanks for the PR! I'm a bit wondering about the underlying issue -- I just tested the code in it's current state in Windows Terminal and it works fine for me.

What version of the terminal are you using? Is Sixel actually working with your change? AFAIK the same version that introduced Sixel support also introduced support for CSI 16 t.

We can't remove the escape sequences. This would break the functionality in other terminals.
Falling back to 10x20 could make sense, basically not erroring out when the escape fails, but using that size. This is how it was a while ago in the code. But as output quality can be quite bad in this case I decided to have a clear error and implement whatever is supported in other terminals, especially as I don't know about a terminal that supports Sixel/TGP but don't understand CSI 16 t.

Can you provide some more information about you setup? Then I'll have a look myself and we can see if there's another way to get the cell size, or, alternatively, go with the fallback, if there isn't a way.

Vedant-Asati03 commented:
> Thanks for the PR! I'm a bit wondering about the underlying issue -- I just tested the code in it's current state in Windows Terminal and it works fine for me.
> 
> What version of the terminal are you using? Is Sixel actually working with your change? AFAIK the same version that introduced Sixel support also introduced support for CSI 16 t.
> 
> We can't remove the escape sequences. This would break the functionality in other terminals. Falling back to 10x20 could make sense, basically not erroring out when the escape fails, but using that size. This is how it was a while ago in the code. But as output quality can be quite bad in this case I decided to have a clear error and implement whatever is supported in other terminals, especially as I don't know about a terminal that supports Sixel/TGP but don't understand CSI 16 t.
> 
> Can you provide some more information about you setup? Then I'll have a look myself and we can see if there's another way to get the cell size, or, alternatively, go with the fallback, if there isn't a way.

Thanks for replying so quickly, can you please tell me which windows terminal you used, regular or preview? Because I found out something interesting while I was using **Regular Windows Terminal** it threw that `Failed to get cell size` error in both versions `1.21.2911.0`[latest stable release] and `1.22.2912.0`[beta release] but when I switched to **Windows Terminal Preview** it was working as intended in both versions.
This is all without implementing the changes I made, and after incorporating the changes I made the code was still working just fine. And yes `Sixel` is working, even without the changes I made.

lnqs commented:
[As mentioned in the README](https://github.com/lnqs/textual-image?tab=readme-ov-file#windows-terminal), Sixel support was added in version 1.22. Which currently is the preview version. That version added both CSI 16 t and the capability to render Sixels. So I didn't test it on earlier version as it's clear it won't work.
One could argue the unicode fallbacks would work in other versions if we wouldn't throw an exception but use a fallback size, but I guess no one is using this package for that functionality. But I'm open to change that behaviour if someone comes up with the actual need/use cases for this.
